aws-api/src/test/java/com/amplifyframework/api/aws/GsonAWSDateTimeTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+import com.amplifyframework.api.aws.internal.AWSDateTimeTypeAdapter;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.SimpleTimeZone;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test cases to test that GSON deserialization process can correctly
+ * handle AWSDateTime format, which DOES NOT exactly follow the
+ * standard ISO-8601 format.
+ *
+ * The main difference is that AWSDateTime can also support seconds
+ * field inside TimeZone offset, which will cause a crash in default
+ * deserializer in GSON if present.
+ */
+@SuppressWarnings("MagicNumber")
+public class GsonAWSDateTimeTest {
+    /*
+     * PST on my local machine at the time of writing this,
+     * but it should vary based on the machine.
+     */
+    private static final TimeZone DEFAULT_TIME_ZONE = TimeZone.getDefault();
+
+    private static Calendar cal;
+    private static Gson gson;
+
+    /**
+     * Configure gson to use the correct custom type adapter for Date class.
+     */
+    @BeforeClass
+    public static void setup() {
+        gson = new GsonBuilder()
+                .registerTypeAdapter(Date.class, new AWSDateTimeTypeAdapter())
+                .create();
+    }
+
+    /**
+     * Test that a string containing just date can be parsed to Date.
+     *
+     * Amplify will assume local timezone if none is specified. This
+     * means that time (in UNIX epoch) is ACTUALLY being affected by
+     * the local timezone of the machine that is parsing the string.
+     */
+    @Test
+    public void testDateWithoutOffset() {
+        cal = new GregorianCalendar(DEFAULT_TIME_ZONE);
+        resetFields(cal);
```

Comment:
Can you do [`cal.setTimeInMillis(0)`](https://docs.oracle.com/javase/7/docs/api/java/util/Calendar.html#setTimeInMillis(long))?

Reply:
im dumb. you are absolutely right